controlplane/kubeadm/controllers/scale.go

```diff
@@ -84,6 +92,11 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 ) (ctrl.Result, error) {
 	logger := controlPlane.Logger()
 
+	// run preflight checks ensuring the control plane is stable before proceeding with a scale up/scale down operation; if not, wait.
```

Comment:
```suggestion
	// Run preflight checks to ensure that the control plane is stable 
	// before proceeding with a scale up/scale down, or init operations; otherwise, wait.
```